incubator/hnc/internal/reconcilers/anchor.go

```diff
@@ -118,13 +119,19 @@ func (r *AnchorReconciler) onDeleting(ctx context.Context, log logr.Logger, inst
 		return false, err
 	}
 
+	// Update the state of the anchor
+	//
+	// TODO: call this in the parent function after v0.5 so there's no special code path for the
+	// onDeleting case.
+	r.updateState(log, inst, snsInst)
```

Comment:
I'm not sure if you added the `updateState()` here just to make the change small in v0.5. It seemed the same for me if just moving the existing `updateState()` up before `onDeleting()` in the `Reconcile()`.

Reply:
Yes, I wanted to make this change as localized as possible in v0.5. In the master branch that I'm working on, I move this to Reconcile.